src/test.py

```diff
@@ -88,10 +136,44 @@ def add_markers(view, response):
             # Only add pass for line if there's no fail for the same line.
             if line not in results["fail"]:
                 results["pass"][line] = {
+                    "name": result[edn.Keyword("var-meta")][edn.Keyword("name")],
                     "type": result[edn.Keyword("type")],
                     "region": sublime.Region(point, point),
                 }
 
+    return results
+
+
+def serializable_results(results):
+    """
+    Returns a list of results suitable for persistence.
+
+    A result is a dict with keys "name", "type", and "region".
+
+    Where:
+        - "name" is the name of the test var;
+        - "type" is one of "pass", "fail", or "error"
+        - "region" is encoded as a tuple(begin, end).
+    """
+
+    def serialize(result):
+        return {
+            "name": result["name"].name,
+            "type": result["type"].name,
+            "region": (result["region"].begin(), result["region"].end())
+        }
+
+    results_ = []
+
+    for result_type in ["pass", "fail", "error"]:
+        results_.extend([serialize(result) for result in results[result_type].values()])
+
+    return results_
```

Comment:
I think we can simplify this to something like this:

```python
    return [serialize(result)
            for result_type in ["pass", "fail", "error"]
            for result in results[result_type].values()]
```

Reply:
Niice!